### PR TITLE
Minor bugfixes and improvements to properties calculation.

### PR DESCRIPTION
**Pull request by fedepedron**
Changed mulliken_calc so it only needs one variable for charges, and fixed the routine calls accordingly. Also minor related changes to other routines in properties and liomain.

### CODE REVIEW

File: lioamber/properties.f90 [hunk 1]
```diff
@@ -73,25 +74,20 @@ end subroutine get_degeneration
 !%% MULLIKEN_CALC %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%!
 ! Performs a Mulliken Population Analysis and outputs atomic charges.          !
 !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%!
-subroutine mulliken_calc(N, M, RealRho, Smat, NofM, q0, q)
+subroutine mulliken_calc(N, M, RealRho, Smat, NofM, q)
 
     ! RealRho         : Rho written in atomic orbitals.                !
     ! q0, q           : Starting and final Mulliken charges.           !
```
Comment (manuelF) [inline lioamber/properties.f90]: Remove q0  and update the comment now that it no longer exists

File: lioamber/properties.f90 [hunk 2]
```diff
@@ -29,6 +29,7 @@ subroutine get_degeneration(energ, orb, M, nDeg, nDegMO)
          
     if (M.eq.1) then
         nDeg = 1
+        nDegMO(1) = 1
```
Comment (manuelF) [inline lioamber/properties.f90]: Why this change?

Comment (fedepedron) [inline lioamber/properties.f90]: That is because the get_degeneration routine returns nDeg (degeneration number) and nDegMO, which are the orbitals degenerated. If M=1, then nDeg=1 but also nDegMO should be set to 1.

Comment (manuelF) [inline lioamber/properties.f90]: Change looks unrelated to the rest of the PR (which is just functional changes) and are non-obvious. Either make a separate PR or add comments in this if block for what it is doing.

Comment (fedepedron) [inline lioamber/properties.f90]: I believe it is related, since it is indeed a bugfix to a property calculation. If this is not implemented, nDegMO(1) will return 0 which is not correct.
